prov/sockets/src/sock_av.c

```diff
@@ -612,24 +571,21 @@ int sock_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 	default:
 		SOCK_LOG_ERROR("Invalid address format: only IPv4 supported\n");
 		ret = -FI_EINVAL;
-		goto err3;
+		goto err2;
 	}
 	_av->rx_ctx_bits = attr->rx_ctx_bits;
 	_av->mask = attr->rx_ctx_bits ?
 		((uint64_t)1 << (64 - attr->rx_ctx_bits)) - 1 : ~0;
 	*av = &_av->av_fid;
 	return 0;
 
-err_shm:
-	shm_unlink(_av->name);
-	ret = -FI_EINVAL;
-	goto err2;
-err3:
-	free(_av->table_hdr);
 err2:
-	free(_av->name);
-err1:
-	free(_av);
-
+	if(attr->name) {
+		ofi_shm_unmap(&_av->shm);
+	} else {
+		if(_av->table_hdr != MAP_FAILED)
```

Comment:
I dont think we need this check here, attr->name = NULL indicates table_hdr to be either NULL or valid ptr.